active-rfcs/0000-tables.md

```diff
@@ -0,0 +1,238 @@
+Each cell has an `id`, `x`, `y` and dimensions. Since when resizing the width and height of cells can be altered hence recording the dimensions of each cell.
+
+The `row` and `column` helps in identifying the row and column indexes of the cell.
+
+Each cell has a `boundElements` to keep it in sync with text containers / labeled arrows, however we can surely rename it to `textId` to keep it simple.
+
+## Adding and Deleting rows / columns
+
+To start with we can keep it simple and just allow to add
+3 x 3 grid table. Once table is added we can have a add row and add column button to increase the rows and columns similar to how **Miro** has.
+
+![uploaded image](https://i.imgur.com/X21LzGR.png)
+
+Similarly and entire `row` and `column` can be removed by right clicking a cell. We just set the attribute `isDeleted` to true for all cells in that row / column and since we are storing the `row` and `column` index in each cell, it will be easy to identify which cells to remove.
+
+## Interacting with tables
+
+Similar to other shape tool, the table tool will be resizable, movable etc.
+
+Whenever user clicks on the cell - show a text box to updated or add the text. The text will start wrapping as per the cell dimensions similar to how it works for text container.
+
+Additionally if a user clicks on the cell stroke (width / height) should be adjustable. So we need to do a hit test whether the cell stroke is being hit and allow the user to adjust dimensions.
+But since this would be internally using rectangles so this should be possible without custom code.
+What we need to support is allow the width / height of all cells in that row /column when any one of them is moved.
+
+Since the table is not just a single element but a collection of different elements, whenever there is an operation eg moving , resizing, we need to update all the children of the table (basically every cell).
+
+## Supporting row and column headers
+
+Ideally the first row and column should be preserved for the row and column header. But do we need a separate distinction for headers ? (Eg showing header in diff background color or some highlighter how miro does it).
+
+I think the users will be able to style the headers differently once we have support for wyswyg editor, so till then lets keep it simple.
+
+## Alternatives
+
+In the above approach the text elements with `containerId` as `cell id` will exists in the `json`. This means there will be 4 separate `text` elements in the `json`.
+
+Here is an alternative version - Having `cells` as virtual text elements instead of actual text elements.
+
+```js
+{
+  "id": "table-id",
+  "type": "table",
+  "x": 100,
+  "y": 100,
+  "title": "Table Title",
+  "rows": 2,
+  "columns": 2,
+  "cells": [
+    {
+      "id": "cell-1",
+      "row": 0,
+      "column": 0,
+      "x": 100,
+      "y": 100,
+      "width": 100,
+      "height": 50,
+      "text":"Cell 1",
+      "fontSize": 16;
+      "fontFamily": 1;
```

Comment:
@dwelle, as per my understanding, can you confirm if this is what you meant by `virtual` elements? These elements would be drawn at the time of rendering so there will be no existence of actual separate text elements